livingatlas/pipelines/src/main/java/au/org/ala/pipelines/beam/ALAUUIDMintingPipeline.java

```diff
@@ -107,16 +131,17 @@ public static void run(InterpretationPipelineOptions options) throws Exception {
     // lookup collectory metadata for this data resource
     ALACollectoryMetadata collectoryMetadata = dataResourceKvStore.get(options.getDatasetId());
     if (collectoryMetadata.equals(ALACollectoryMetadata.EMPTY)) {
-      throw new RuntimeException(
-          "Unable to retrieve dataset metadata for dataset: " + options.getDatasetId());
+      log.error("Unable to retrieve dataset metadata for dataset: " + options.getDatasetId());
+      System.exit(1);
     }
 
     // construct unique list of darwin core terms
-    final List<String> uniqueTerms =
-        collectoryMetadata.getConnectionParameters().getTermsForUniqueKey();
-    if (uniqueTerms == null || uniqueTerms.isEmpty()) {
-      throw new RuntimeException(
-          "No unique terms specified for dataset: " + options.getDatasetId());
+    List<String> uniqueTerms = collectoryMetadata.getConnectionParameters().getTermsForUniqueKey();
+
+    if ((uniqueTerms == null || uniqueTerms.isEmpty())) {
+      log.error(
+          "Unable to proceed, No unique terms specified for dataset: " + options.getDatasetId());
+      System.exit(1);
     }
```

Comment:
Just checking that a System.exit, rather than something that communicates to higher layers is what's expected here and not a left-over artefact from debugging.

Reply:
It is expected but not desirable. Theres an open issue here:

https://github.com/AtlasOfLivingAustralia/la-pipelines/issues/105